### ISSUE TITLE
Will Gooey support automatic settings from inspect.getfullargspec(function) and type hint

### ISSUE THREAD

**Issue by ClericPy** [chriskiehl/Gooey#493]
`inspect.getfullargspec` will get some meta info
Such as 
- arg type annotations
- arg required
- return type annotations
- complex type like typing.Dict[str, int]

(`inspect.signature` is better than `inspect.getfullargspec`)

Is there a possibility to generate the settings automaticly. This operation will save a lot of time from **GooeyParser.add_argument**.
```python
import inspect
import typing


def test1(a, b: int, c: str = '3', *,
          d: typing.Dict[str, int]) -> typing.Dict[str, str]:
    return {'test': 'ok'}


# FullArgSpec(args=['a', 'b', 'c'], varargs=None, varkw=None, defaults=('3',), kwonlyargs=['d'], kwonlydefaults=None, annotations={'return': typing.Dict[str, str], 'b': <class 'int'>, 'c': <class 'str'>, 'd': typing.Dict[str, int]})
print(inspect.getfullargspec(test1))


def test2(a,
          b: int,
          c: str = '3',
          *args: typing.List[str],
          **kwargs: typing.Dict[str, int]) -> typing.Dict[str, str]:
    return {'test': 'ok'}


# FullArgSpec(args=['a', 'b', 'c'], varargs='args', varkw='kwargs', defaults=('3',), kwonlyargs=[], kwonlydefaults=None, annotations={'return': typing.Dict[str, str], 'b': <class 'int'>, 'c': <class 'str'>, 'args': typing.List[str], 'kwargs': typing.Dict[str, int]})
print(inspect.getfullargspec(test2))

```

Sorry for my **Plastic English**, and thanks for the awesome lib which is great useful. Perhaps there may be the Web UI and Terminal UI version.

**Comment by chriskiehl** [comment]
I'm not totally sure I understand what you're proposing. `inspect` is some magic with which I am not familiar. Could you maybe translate one of the [Gooey examples](https://github.com/chriskiehl/GooeyExamples) using the argspec thing? 

For a Web version, look no further than Wooey! https://github.com/wooey/Wooey

**Comment by ClericPy** [comment]
@chriskiehl 
Yes, I found Wooey after opening this issue, and that is enough for using.

It is just a suggestion.

Instead of manually filling args with `parser.add_argument` and generating it, I'm trying to generate an entire GUI program by passing in only one function object as an argument. Thanks to Python's powerful introspection, all args that needed will be gotten from `inspect.signature`. Besides the argument names, the type hints will give a hand for translating string into correct-type object. Just like 
[pydantic](https://github.com/samuelcolvin/pydantic) do.

I tried a lot for automately generating the GUI / Terminal UI of function with inspect.signature & pydantic. It is terriblely difficult to deal with plenty of compatibility bugs and type conversion. So for performance and stability reason, Gooey should not include this feature.

I found this may not be a very smart issue, which may need to be closed. Thanks for your help again.
